substrate/frame/broker/src/adapt_price.rs

```diff
@@ -17,59 +17,114 @@
 
 #![deny(missing_docs)]
 
-use crate::CoreIndex;
+use crate::{CoreIndex, SaleInfoRecord};
 use sp_arithmetic::{traits::One, FixedU64};
-use sp_runtime::Saturating;
+use sp_runtime::{FixedPointNumber, FixedPointOperand, Saturating};
+
+/// Performance of a past sale.
+#[derive(Copy, Clone)]
+pub struct SalePerformance<Balance> {
+	/// The price at which the last core was sold.
+	///
+	/// Will be `None` if no cores have been offered.
+	pub sellout_price: Option<Balance>,
+
+	/// The minimum price that was achieved in this sale.
+	pub min_price: Balance,
+
+	/// The number of cores we want to sell, ideally.
+	pub ideal_cores_sold: CoreIndex,
+
+	/// Number of cores which are/have been offered for sale.
+	pub cores_offered: CoreIndex,
+
+	/// Number of cores which have been sold; never more than cores_offered.
+	pub cores_sold: CoreIndex,
+}
+
+/// Result of `AdaptPrice::adapt_price`.
+#[derive(Copy, Clone)]
+pub struct AdaptedPrices<Balance> {
+	/// New minimum price to use.
+	pub min_price: Balance,
+	/// Price we optimize for.
+	pub target_price: Balance,
```

Comment:
Really this is the market renewal price, but that's not clear here. What does it mean that we optimise for this price?

Reply:
Optimize for in the sense that we adjusted the curve to handle swings around that price optimally, but your are right this is more an implementation detail - it is supposed to be used for renewals, let's keep the old name.

Reply:
actually this is also misleading as this is only renewals of legacy leases :thinking:

Reply:
Improved docs, but left name.

Reply:
After our conversation the other day I had thought it was a good idea to allow the price adapter to influence normal renewals, wdyt?